From 0ace034ab77bc9ed81e129d2811ddc6e5deb892d Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Sat, 11 May 2024 13:29:36 +0800
Subject: [PATCH 041/117] vavp9enc: Do not use base class video info to
 calculate coded size

We should use our in_info which is an adjusted value to calculate
that coded size.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/6826>
---
 sys/va/gstvavp9enc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sys/va/gstvavp9enc.c b/sys/va/gstvavp9enc.c
index 5c8817d2d..15968b5c5 100644
--- a/sys/va/gstvavp9enc.c
+++ b/sys/va/gstvavp9enc.c
@@ -1752,8 +1752,8 @@ _vp9_calculate_coded_size (GstVaVp9Enc * self)
       codedbuf_size = codedbuf_size / 2;
   } else if (self->rc.max_bitrate_bits > 0) {
     guint64 frame_sz = gst_util_uint64_scale (self->rc.max_bitrate_bits / 8,
-        GST_VIDEO_INFO_FPS_D (&base->input_state->info),
-        GST_VIDEO_INFO_FPS_N (&base->input_state->info));
+        GST_VIDEO_INFO_FPS_D (&base->in_info),
+        GST_VIDEO_INFO_FPS_N (&base->in_info));
 
     /* FIXME: If average frame size is smaller than 1/10 coded buffer size,
        we shrink the coded buffer size to 1/2 to improve performance. */
-- 
2.43.0

